libbeat/common/bus/bus.go

```diff
@@ -0,0 +1,102 @@
+package bus
+
+import (
+	"sync"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// Event sent to the bus
+type Event common.MapStr
+
+// Bus provides a common channel to emit and listen for Events
+type Bus interface {
+	// Emit an event to the bus
+	Emit(Event)
+
+	// Subscribe to all events, filter them to the ones containing *all* the keys in filter
+	Subscribe(filter ...string) Listener
+}
+
+// Listener retrieves Events from a Bus subscription until Close is called
+type Listener interface {
+	// Events channel
+	Events() <-chan Event
+
+	// Stop listening and removes itself from the bus
+	Stop()
+}
+
+type bus struct {
+	sync.RWMutex
+	name      string
+	listeners []*listener
+}
+
+type listener struct {
+	filter  []string
+	channel chan Event
+	bus     *bus
+}
+
+// New initializes a new bus with the given name and returns it
+func New(name string) Bus {
```

Comment:
What are the names of the buses you have in mind?

Reply:
I don't know, this name is used just for logging